gateway-ha/src/main/java/io/trino/gateway/proxyserver/RouteToBackendResource.java

```diff
@@ -60,11 +60,12 @@ public void postHandler(
             @Context HttpServletRequest servletRequest,
             @Suspended AsyncResponse asyncResponse)
     {
-        if (servletRequest.getRequestURI().startsWith(V1_STATEMENT_PATH)) {
+        MultiReadHttpServletRequest multiReadHttpServletRequest = new MultiReadHttpServletRequest(servletRequest, body);
+        if (multiReadHttpServletRequest.getRequestURI().startsWith(V1_STATEMENT_PATH)) {
             proxyHandlerStats.recordRequest();
         }
-        String remoteUri = routingTargetHandler.getRoutingDestination(servletRequest);
-        proxyRequestHandler.postRequest(body, servletRequest, asyncResponse, URI.create(remoteUri));
+        String remoteUri = routingTargetHandler.getRoutingDestination(multiReadHttpServletRequest);
+        proxyRequestHandler.postRequest(body, multiReadHttpServletRequest, asyncResponse, URI.create(remoteUri));
```

Comment:
Why would this apply only to POST and not to others GET, DELETE?
Would it not be safer to use MultiReadHttpServletRequest instead of HttpServletRequest in all cases?
including RoutingGroup codes?

Reply:
the GETs and DELETEs won't have bodies, so `getReader` and `getInputStream` will be null either way

Reply:
True, hmm.. how about using one type just for unifcation for all cases? even if it doesn't have body?

Reply:
Yeah, I think that is what @oneonestar was suggesting with the plan to replace `HttpServeletRequest` with a dedicated class like `TrinoRequestContext`. This is the way to go, but would broaden the scope of this PR by quite a bit. So for now I think we can get by with `HttpServletRequest`, using the MultiRead implementation when there is a body